src/ConnectionAutomaton.cpp

```diff
@@ -241,15 +297,44 @@ void ConnectionAutomaton::cancel() {
 }
 /// @}
 
+void ConnectionAutomaton::scheduleKeepalive(const uint32_t keepAliveDelay) {
+  if (keepAliveDelay > 0) {
+    auto eventBase = folly::EventBaseManager::get()->getExistingEventBase();
+    CHECK(eventBase);
+
+    eventBase->runAfterDelay(
+        [this, keepAliveDelay]() { sendKeepalive(keepAliveDelay); },
```

Comment:
is it possible that this instance will be destroyed when the eventBase will execute the callback?
I believe that its possible. In that case we need to capture shared_ptr<ConnectionAutomaton> thisPtr.